### PR TITLE
CSHARP-5035: Improve serverless tests to obtain secrets from vault

### PR DESCRIPTION
**Pull request by sanych-sun**

### CODE REVIEW

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -738,9 +738,6 @@ functions:
           - "FLE_AZURE_CLIENT_SECRET"
           - "FLE_GCP_EMAIL"
           - "FLE_GCP_PRIVATE_KEY"
-          - "SERVERLESS_ATLAS_USER"
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:741]: Credentials is being fetched from the vault now.

File: evergreen/evergreen.yml [hunk 2]
```diff
@@ -769,47 +766,6 @@ functions:
           FRAMEWORK=${FRAMEWORK} \
             evergreen/run-tests.sh
 
-  create-serverless-instance:
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:772]: I've deleted `create-serverless-instance` and `delete-serverless-instance-if-configured` tasks by moving them into `setup` and `teardown` of `serverless-task-group`.

File: evergreen/evergreen.yml [hunk 3]
```diff
@@ -1059,9 +1015,6 @@ pre:
   - func: make-files-executable
 
 post:
-  # Removed, causing timeouts
-  # - func: upload-working-dir
-  - func: delete-serverless-instance-if-configured
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:1064]: Do not need it here, because serverless tests have it's own task group now.

File: evergreen/evergreen.yml [hunk 4]
```diff
@@ -1279,7 +1232,6 @@ tasks:
     - name: test-serverless
       exec_timeout_secs: 2700 # 45 minutes: 15 for setup + 30 for tests
       commands:
-        - func: create-serverless-instance
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:1282]: Moved to `setup` of `serverless-task-group`.

File: evergreen/evergreen.yml [hunk 5]
```diff
@@ -2387,17 +2368,12 @@ buildvariants:
     - name: "test-load-balancer-netstandard21"
 
 # Serverless tests
-- matrix_name: serverless-tests-windows
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:2390]: Some of evergreen-tools scripts were not happy with Windows host (I've got error: `jq: command not found`). I think it should be enough to run serverless tests on Ubuntu only. But if it's vital for us to have Windows tests as well - I can dig deeper to understand what we can do.

**Review by BorisDog — APPROVED** [review]
LGTM, Good refactoring!
Just need to decide re. windows.

**Pull request merged**